pkg/record/record.go

```diff
@@ -12,34 +12,87 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:generate stringer -type=Operation -trimprefix Operation
+
 package record
 
 import (
 	"encoding/json"
-	"time"
+	"fmt"
 
 	"github.com/conduitio/conduit/pkg/foundation/cerrors"
 	"github.com/conduitio/conduit/pkg/record/schema"
 )
 
-// Record ...
+const (
+	OperationCreate Operation = iota + 1
+	OperationUpdate
+	OperationDelete
+	OperationSnapshot
+)
+
+// Operation defines what triggered the creation of a record.
+type Operation int
+
+// Record represents a single data record produced by a source and/or consumed
+// by a destination connector.
 type Record struct {
```

Comment:
We now have essentially the same code for the record in the SDK and in Conduit (see [this](https://github.com/ConduitIO/conduit-connector-sdk/blob/main/record.go#L52)). According to @ahmeroxa they want to add a similar / the same struct to Turbine as well. If we all agree to use the OpenCDC record it would make sense to extract this into a repository that contains the OpenCDC protobuf definition, the Go struct and utilities to transform it into the protobuf definition, then import it as a library in the SDK and Conduit.

Reply:
I like the idea. :+1: